### PR TITLE
Several documentation adjustments for minor features/enhancements

### PR DESCRIPTION
**Pull request by smcvb**
This pull request introduces a bunch of minor additions/changes following from the new features in Axon Framework 4.3:

- [Expose remaining replay 'duration' in Tracking Processor status](https://github.com/AxonFramework/AxonFramework/issues/1252)
- [Expose 'merging status' in Tracking Processor status](https://github.com/AxonFramework/AxonFramework/issues/1253)
- [Improve customizability of SQL statements in JdbcEventStorageEngine](https://github.com/AxonFramework/AxonFramework/issues/1187)
- [Allow disabling AxonServer AutoConfiguration through property](https://github.com/AxonFramework/AxonFramework/pull/1334)
- [Introduce an EventTypeUpcaster](https://github.com/AxonFramework/AxonFramework/pull/1310)
- [Register Parameter Resolvers in Test Fixtures](https://github.com/AxonFramework/AxonFramework/issues/1050)
- [Register Handler Enhancer Definitions in Test Fixtures](https://github.com/AxonFramework/AxonFramework/issues/1049)
- [Expect Scheduled Deadline based on Deadline name](https://github.com/AxonFramework/AxonFramework/issues/680)
- [Test fixture API to check for scheduled event / deadline cancellation](https://github.com/AxonFramework/AxonFramework/issues/691)
- [Cancel Deadline within Scope](https://github.com/AxonFramework/AxonFramework/issues/1235)

### CODE REVIEW

File: configuring-infrastructure-components/deadlines.md [hunk 1]
```diff
@@ -33,18 +39,39 @@ Cancelling a deadline could for example come in handy when a certain event means
...
+    void cancelMyDeadline(String deadlineId) {
+        deadlineManager.cancelSchedule("myDeadline", deadlineId);
+    }
+}
 ```
 
-> **Canceling Deadlines**
->
-> It is possible to cancel all deadlines of a given name by invoking `deadlineManager.cancelAll("myDeadline")`.
+There are more option to cancel a deadline nexted to through the received `deadlineId` and the used `deadlineName`:
```
Comment (lfgcampos) [inline configuring-infrastructure-components/deadlines.md]:
```suggestion
There are more options to cancel a deadline than the previously mentioned:
```

File: configuring-infrastructure-components/deadlines.md [hunk 2]
```diff
@@ -33,18 +39,39 @@ Cancelling a deadline could for example come in handy when a certain event means
...
 
-If you need some contextual data about the Deadline during the Deadline Handling,
- you can attach a Deadline Payload when scheduling a Deadline:
+ * `cancelAll(String)`
```
Comment (lfgcampos) [inline configuring-infrastructure-components/deadlines.md]:
```suggestion
 * `cancelAll(String name)`
```

File: configuring-infrastructure-components/deadlines.md [hunk 3]
```diff
@@ -33,18 +39,39 @@ Cancelling a deadline could for example come in handy when a certain event means
...
+  Cancels _every_ scheduled deadline matching the given deadline name.
+  Note that this thus also cancels deadlines from other aggregate and/or saga instances matching the name.
+ * `cancelAllWithinScope(String)`
```
Comment (lfgcampos) [inline configuring-infrastructure-components/deadlines.md]:
```suggestion
 * `cancelAllWithinScope(String name)`
```

File: configuring-infrastructure-components/deadlines.md [hunk 4]
```diff
@@ -33,18 +39,39 @@ Cancelling a deadline could for example come in handy when a certain event means
...
+  Cancels a scheduled deadline matching the given deadline name, _within_ the scope the method is invoked in.
+  For example, if this operation is performed from within "aggregate instance X",
+   the `ScopeDescriptor` from "aggregate instance X" will be used to cancel. 
+ * `cancelAllWithinScope(String, ScopeDescriptor)`
```
Comment (lfgcampos) [inline configuring-infrastructure-components/deadlines.md]:
```suggestion
 * `cancelAllWithinScope(String name, ScopeDescriptor scope)`
```

File: operations-guide/production-considerations/monitoring-and-metrics.md [hunk 5]
```diff
@@ -75,11 +75,21 @@ To that end, the `TrackingEventProcessor` exposes the `processingStatus()` metho
...
+ * A Boolean through `isCaughtUp()` specifying whether it is caught up with the Event Stream.
+ * A Boolean through `isReplaying()` specifying whether the given Segment is
+  [replaying](../../configuring-infrastructure-components/event-processing/event-processors.md#replaying-events).
+ * A Boolean through `isMerging()` specifying whether the given Segment is
+  [merging](../../configuring-infrastructure-components/event-processing/event-processors.md#splitting-and-merging-tracking-tokens).
  * The `TrackingToken` of the given Segment.
- * A Boolean specifying whether the Segment is in an error state.
+ * A Boolean through `isErrorState()` specifying whether the Segment is in an error state.
  * An optional `Throwable` if the Event Tracker reached an error state.
+ * An optional `Long` through the  defining the current position of the `TrackingToken`.
```
Comment (lfgcampos) [inline operations-guide/production-considerations/monitoring-and-metrics.md]:
```suggestion
 * An optional `Long` through `getCurrentPosition()` defining the current position of the `TrackingToken`.
```

File: operations-guide/production-considerations/monitoring-and-metrics.md [hunk 6]
```diff
@@ -75,11 +75,21 @@ To that end, the `TrackingEventProcessor` exposes the `processingStatus()` metho
...
+ * An optional `Long` defining the position at reset of the `TrackingToken`.
```
Comment (lfgcampos) [inline operations-guide/production-considerations/monitoring-and-metrics.md]:
```suggestion
 * An optional `Long` through `getResetPosition()` defining the position at reset of the `TrackingToken`. 
```

File: operations-guide/production-considerations/monitoring-and-metrics.md [hunk 7]
```diff
@@ -75,11 +75,21 @@ To that end, the `TrackingEventProcessor` exposes the `processingStatus()` metho
...
+  [merging](../../configuring-infrastructure-components/event-processing/event-processors.md#splitting-and-merging-tracking-tokens).
  * The `TrackingToken` of the given Segment.
- * A Boolean specifying whether the Segment is in an error state.
+ * A Boolean through `isErrorState()` specifying whether the Segment is in an error state.
  * An optional `Throwable` if the Event Tracker reached an error state.
+ * An optional `Long` through the  defining the current position of the `TrackingToken`.
+ * An optional `Long` defining the position at reset of the `TrackingToken`. 
+  This field will be `null` in case the `isReplaying()` returns `false`.
+  It is possible to derive an estimated duration of replaying by comparing the current position with this field.
+ * An optional `Long` defining the position on the `TrackingToken` when merging will be completed.
```
Comment (lfgcampos) [inline operations-guide/production-considerations/monitoring-and-metrics.md]:
```suggestion
 * An optional `Long` throug `mergeCompletedPosition()` defining the position on the `TrackingToken` when merging will be completed. 
```

File: configuring-infrastructure-components/deadlines.md (hunk 1 above)
Comment (smcvb) [reply configuring-infrastructure-components/deadlines.md]: Thanks for sharing.

File: configuring-infrastructure-components/deadlines.md (hunk 3 above)
Comment (smcvb) [reply configuring-infrastructure-components/deadlines.md]: I'll add the parameter name, good point. I'll do `deadlineName` though, to be a tad more specific.

**Pull request merged**